PHASE_1/API_SourceCode/src/api/routes/ReportSpecific.router.ts

```diff
@@ -0,0 +1,49 @@
+import { Router, Request, Response, NextFunction } from "express";
+import { formatError, getLogger } from "../../utils/Logger";
+import { IRouter } from "../../interfaces/IRouter";
+import { ReportService } from "../services/Report.service";
+import validationMiddleware from "../middlewares/validation";
+import { ReportSchema } from "../schemas/Report.schema";
+import { parseReportHeaders } from "../../utils/Helpers";
+import { IReportRequestHeaders } from "IApiResponses";
+
+export class ReportSpecificRouter implements IRouter {
+  private readonly logger = getLogger();
+  private readonly router: Router;
+  private readonly prefix = "/api/v1";
+
+  constructor(private readonly reportService: ReportService) {
+    this.router = this.setupRoutes();
+  }
+
+  setupRoutes(): Router {
+    return Router().get(
```

Comment:
instead of entire new router, just take this route and add to ReportRouter

Reply:
No clue how to go about this sorry, everything I've tried is way off haha could I get a hint?